colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -296,4 +300,13 @@ extends Controller[O,I](codec, ControllerConfig(config.pendingBufferSize, config
     }
   }
 
+
+  override def idleCheck(period: Duration) {
+    super.idleCheck(period)
+
+    if (sentBuffer.size > 0 && sentBuffer.front.isTimedOut(System.currentTimeMillis)) {
+      // the oldest sent message has expired with no response - shutdown the connection
+      disconnect()
```

Comment:
Ok there is a bug with `connectionStatus` returning the wrong value, but the client isn't actually trying to reconnect.

I'm just going to push the fix right into this branch because it's really minor and I also made a couple minor edits to your test.